File: iteration2/src/controllers/StudentController.java
package iteration2.src.controllers;

import java.util.List;
import iteration2.src.models.CourseSection;
import iteration2.src.models.SelectedCourse;
import iteration2.src.models.Student;

public class StudentController {
    private Student currentStudent;

    public StudentController(Student currentStudent) {
        this.currentStudent = currentStudent;
    }

    public boolean addSelectedCourse(SelectedCourse selectedCourse) {

        return currentStudent.addNewCourse(selectedCourse);
    }

    public List<CourseSection> getAvailableCourseSections() {
        return currentStudent.listAvailableCourseSections();
    }

    public boolean removeSelectedCourse(SelectedCourse selectedCourse) {
        return currentStudent.deleteCourse(selectedCourse);
    }

    public void sendSelectedCoursesToApproval() {
        currentStudent.sendSelectedCoursesToApproval();
    }

    public List<String> getNotification() {
        return currentStudent.getNotifications();
    }

    public String[][] getTimeTable() {
        return currentStudent.createTimeTable();
    }

    public void logout() {
        System.exit(0);
    }

    public List<CourseSection> getAvaliableCourseSections() {
        return currentStudent.listAvailableCourseSections();
    }

    public List<SelectedCourse> getSelectedCourses() {
        
        return currentStudent.getSelectedCourses();
    }

    public List<String> getNotifications() {
        return currentStudent.getNotifications();
    }

	

    
}

File: iteration3/src/controllers/StudentController.py
#beyz
class StudentController:
    def __init__(self, currentStudent):
        self._currentStudent = currentStudent
        
    def addSelectedCourse(self, selectedCourse):
        return self._currentStudent.addNewCourse(selectedCourse)
    
    def getAvailableCourseSections(self):
        return self._currentStudent.listAvailableCourseSections()
    
    def removeSelectedCourse(self, selectedCourse):
        return self._currentStudent.deleteCourse(selectedCourse)
    
    def sendSelectedCoursesToApproval(self):
        return self._currentStudent.sendSelectedCoursesToApproval()

    def getNotifications(self):
        return self._currentStudent.getNotifications()
    
    def getTimeTable(self):
        return self._currentStudent.createTimeTable()
    
    def logout(self):
        self._currentStudent.logout()

    def clearNotifications(self):
        return self._currentStudent.clearNotifications()
    
    def getSelectedCourses(self):
        return self._currentStudent.getSelectedCourses()
    
    def getApprovalStatus(self):
        return self._currentStudent.getApprovalStatus()
